testing.py

```diff
@@ -398,6 +400,22 @@ def _collection(self, name=u"Faketown Public Library"):
             client_id=u"abc", client_secret=u"def"
         )[0]
 
+    def _subject(self, type, identifier, genre=None):
+        if genre == None:
+            return get_one_or_create(
+                self._db, Subject, type=type, identifier=identifier
+            )[0]
+
+        return get_one_or_create(
+            self._db, Subject, type=type, identifier=identifier, genre=genre
+        )[0]
+
```

Comment:
I don't think you get to specify genre when creating a subject. That's up to the classifier. I see what you're doing in the test but I would take this out of the _subject() helper function because that genre is going to get immeidately overridden by whatever the classifier says.

Within the test below, it's okay to set Subject.genre to a dummy value after creating the Subject object, but I think putting it in the helper function makes a promise that can't be kept.